Lib/importlib/_bootstrap.py

```diff
@@ -263,7 +263,13 @@ def _requires_builtin_wrapper(self, fullname):
 def _requires_frozen(fxn):
     """Decorator to verify the named module is frozen."""
     def _requires_frozen_wrapper(self, fullname):
-        if not _imp.is_frozen(fullname):
+        if not _imp.is_frozen(fullname) or (
+            # bpo-45272: os.path is not a concrete module, but an importable
+            #            attribute of 'os'. _imp.is_frozen('os.path') returns
+            #            False but it is frozen, so we check the loader here.
+            fullname in sys.modules
+            and sys.modules[fullname].__spec__.loader is FrozenImporter
```

Comment:
@ericsnowcurrently, couldn't we do `loader.origin == 'frozen'` instead? Are we planing to change the value of origin to the actual file or something like that?

Reply:
related bpo: https://bugs.python.org/issue35321

Reply:
> couldn't we do loader.origin == 'frozen' instead?

Let's be explicit about it with the check you have.

Reply:
Well, the reasoning is there some different frozen importer type might be added, and this code might break. That scenario seemed reasonable to me with the complex bootstrapping architecture, but, of course, I lack the experience to make educated guesses :stuck_out_tongue:

Reply:
> some different frozen importer type might be added, and this code might break.

We can deal with it if that ever happens.  The set of conditions here is so specific, yet highly unlikely.  Plus, if the check here somehow breaks in the future, the problem should be fairly obvious.  However, I expect it wouldn't ever be a problem.

(Regardless, the point is moot now as I'm pretty sure we can drop the importlib change.)